### DISCUSSION TITLE
[Help]: Configurable threshold for settled

### DISCUSSION THREAD

**Discussion by richsoni-spotify** [davidjerleke/embla-carousel#738] [Help] [answered]
### Which variants of Embla Carousel are relevant to this feature request?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Feature description

Thanks for all the hard work.  This library has been a pleasure to integrate into our ecosystem.  The only snag we have hit is related to the delay in the settled state.

Attached is a screenshot of our carousel with a duration of 20.  As you can see, the time it takes for our disabled state to fade back in is jarring.  Despite the micromovements the DOM might be still making, it would help us if we could adjust the threshold at which everything is considered "settled" so we can reenable our slides.

https://github.com/davidjerleke/embla-carousel/assets/129987359/d81fd338-62ae-434b-beba-7e8a9b76fc47

### Additional Context

Additional details here...

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues
- [X] I agree to follow this project's Contributing Guidelines for feature requests

**Comment by sarussss** [comment]
Hi @richsoni-spotify 
I think you can try event [select](https://www.embla-carousel.com/api/events/#select)

**Reply by davidjerleke** [reply]
@richsoni-spotify why not as @sarussss mentioned use the [`select`](https://www.embla-carousel.com/api/events/#select) event instead? This will fire as soon as the selected scroll snap changes:

```ts
emblaApi.on('select', () => { /* Do your thing... */ })
```

**Reply by richsoni-spotify** [reply]
Hey @davidjerleke, thanks for sharing this.  I tried to use the `select` handler as suggested, with our setup this seems to emit very shortly after the scroll event starts.  We are trying to have the slide remain in its disabled state while it goes beneath the mouse.  Since the slide has a video that plays when hover is active.

Here is the benchmarking I did:
```
Cycle | onClick to select (ms) | select to settle (ms)
------------------------------------------------------
1     | 14                     | 793
2     | 14                     | 112
3     | 15                     | 1188
4     | 14                     | 1193
5     | 15                     | 1188
6     | 16                     | 1188
7     | 17                     | 188
8     | 16                     | 759
9     | 18                     | 762
```

**Reply by richsoni-spotify** [reply]
Use case is the same as https://github.com/davidjerleke/embla-carousel/discussions/739

Since we need to disable the slides during scrolling, but the settled method is waiting to long to reenable.  It would also be possible to do this ourselves with the onScroll method if we knew how far it was from settling.  Like if we didn't control the threshold of the 'settled' event, but knew how far off we were, so we could manipulate some state in our component.

Does that make sense?

**Reply by davidjerleke** [reply, marked as answer]
@richsoni-spotify I see. You should be able to do something along these lines:

```ts
const settlePixelThreshold = 10

function fireCustomSettle(emblaApi) {
  const { dragHandler, location, target } = emblaApi.internalEngine()
  if (dragHandler.pointerDown()) return // Bail if user is dragging

  const displacement = target.get() - location.get()
  if (Math.abs(displacement) < settlePixelThreshold) {
    // do your thing here
  }
}

emblaApi.on('scroll', fireCustomSettle)
```

However, please note that this will fire multiple times as long as `displacement` is less than `settlePixelThreshold`. You might want to add some sort of check to prevent that. 

Best,
David

**Reply by richsoni-spotify** [reply]
This worked perfectly!  Thank you!!!
